src/test/java/edu/harvard/iq/dataverse/datasetutility/FileSizeCheckerTest.java

```diff
@@ -35,4 +61,38 @@ public void testBytesToHumanReadable() {
         assertEquals(expLongAns, longAns);
     }
 
+    @Test
+    public void testIsAllowedFileSize_throwsOnNull() {
+        assertThrows(NullPointerException.class, () -> {
+            fileSizeChecker.isAllowedFileSize(null);
+        });
+    }
+
+    @ParameterizedTest
+    @ValueSource(longs = { 0L, 999L, 1000L })
+    public void testIsAllowedFileSize_allowsSmallerOrEqualFileSize(Long fileSize) {
+        FileSizeResponse response = fileSizeChecker.isAllowedFileSize(fileSize);
+        assertTrue(response.fileSizeOK);
+    }
+
+    @ParameterizedTest
+    @ValueSource(longs = { 1001L, Long.MAX_VALUE })
+    public void testIsAllowedFileSize_rejectsBiggerFileSize(Long fileSize) {
+        FileSizeResponse response = fileSizeChecker.isAllowedFileSize(fileSize);
+        assertFalse(response.fileSizeOK);
+    }
+
+    @Test
+    public void testIsAllowedFileSize_allowsOnUnboundedFileSize() {
+        // initialize a system config and instantiate a file size checker
+        // ensure that a max filesize is not set
+        FileSizeChecker unboundedFileSizeChecker = new FileSizeChecker(new SystemConfig() {
+            @Override
+            public Long getMaxFileUploadSize() {
+                return null;
+            }
+        });
+        FileSizeResponse response = unboundedFileSizeChecker.isAllowedFileSize(1000L);
```

Comment:
Maybe do one more of these checks with a much larger number?

Reply:
I converted this to a parametrized test with 0L, 1000L, and Long.MAX_VALUE, which should cover the domain applicable to the test.